webpack/assets/javascripts/foreman_tools.js

```diff
@@ -85,19 +85,38 @@ export function initTypeAheadSelect(input) {
 // handle table updates via turoblinks
 export function updateTable(element) {
   let uri = new URI(window.location.href);
-  let search, perPage;
+  let searchTerm, perPage, isSearchForm;
 
   if (element !== undefined) {
-    search = $(element).find('.autocomplete-input').val() || $('#search-form').find('.autocomplete-input').val();
-    if (search !== undefined) {
-      uri.setSearch('search', search.trim());
+    isSearchForm = (element.id === 'search-form');
+    if ((isSearchForm || element.id === 'per_page') && getQueryStringValue(uri, 'page') > 1) {
+      uri.setSearch('page', 1);
+    }
+    if (isSearchForm) {
+      searchTerm = $(element).find('.autocomplete-input').val() || $('#search-form').find('.autocomplete-input').val();
```

Comment:
`isSearchForm` means that `element` has id of `search-form`, so that the two conditions are actually checking for the same thing.